src/js/control-bar/fullscreen-toggle.js

```diff
@@ -21,7 +21,21 @@ class FullscreenToggle extends Button {
   buildCSSClass() {
     return `vjs-fullscreen-control ${super.buildCSSClass()}`;
   }
-
+  /**
+   * Handles Fullscreenchange on the component and change control text accordingly
+   *
+   * @method handleFullscreenChange
+   */
+  handleFullscreenChange() {
+    if (this.player_.isFullscreen()) {
+      this.addClass('vjs-fullscreen');
+      this.controlBar.fullscreenToggle.controlText('Non-Fullscreen');
+    }
+    else {
+      this.removeClass('vjs-fullscreen');
+      this.controlBar.fullscreenToggle.controlText('Fullscreen');
```

Comment:
we should make sure the context for this method is the fullscreen toggle itself, otherwise it can be super confusing.